src/autoschedulers/mullapudi2016/AutoSchedule.cpp

```diff
@@ -2799,30 +3265,45 @@ void Partitioner::generate_group_cpu_schedule(
             if ((iter != stg_estimates.end()) && iter->second.defined()) {
                 if (!seq_var.empty()) {
                     VarOrRVar seq(seq_var, (rvars.find(seq_var) != rvars.end()));
-                    f_handle.reorder(seq, v);
+                    if (t.has_gpu_feature()) {
+                        gpu_tiling.canReorder({seq, v});
+                    } else {
+                        f_handle.reorder(seq, v);
+                        sched.push_schedule(f_handle.name(), g.output.stage_num,
+                                            "reorder(" + seq_var + ", " + var + ")",
+                                            {seq_var, var});
+                    }
+                }
+                if (t.has_gpu_feature()) {
+                    auto parallelized_split = gpu_tiling.can_parallelize(v, iter->second);
+                    if (parallelized_split) {
+                        auto split_vars = *parallelized_split;
+                        inner_dims.emplace_back(split_vars.inner);
+                        outer_dims.emplace_back(split_vars.outer);
+                    }
+                } else {
+                    f_handle.parallel(v);
                     sched.push_schedule(f_handle.name(), g.output.stage_num,
-                                        "reorder(" + seq_var + ", " + var + ")",
-                                        {seq_var, var});
+                                        "parallel(" + var + ")", {var});
                 }
-                f_handle.parallel(v);
-                sched.push_schedule(f_handle.name(), g.output.stage_num,
-                                    "parallel(" + var + ")", {var});
                 def_par = simplify(def_par * iter->second);
             } else {
                 break;
             }
         }
     }
 
-    // Silenced: the user can't really do anything about it,
-    // and it triggers on things like tiny lookup tables
-    //
-    // if (can_prove(def_par < arch_params.parallelism)) {
-    //     user_warning << "Insufficient parallelism for " << f_handle.name() << "\n";
-    // }
+    if (can_prove(def_par < arch_params.parallelism)) {
+        debug(1) << "Insufficient parallelism for " << f_handle.name() << "\n";
+    }
+
+    if (t.has_gpu_feature()) {
+        gpu_tiling.apply(sched);
+    }
 
     // Find the level at which group members will be computed.
-    int tile_inner_index = dims.size() - outer_dims.size() - 1;
+    internal_assert(dims.size() >= outer_dims.size());
+    const auto tile_inner_index = std::max(int(dims.size() - outer_dims.size()) - 1, 0);
```

Comment:
For tensor multiplications, the assertion `dims.size() > outer_dims.size()` fails because the GPU schedule wants to map outer dimensions to `gpu_blocks`. If the assertion is ignored, integer `tile_inner_index` becomes invalid (i.e. value less than zero).

Here, I am trying to clamp the `tile_inner_index` to zero. This may break more test case. Help wanted here.